test/blackbox-tests/test-cases/config/config-version.t

```diff
@@ -0,0 +1,56 @@
+Tests verifying that the config file presents accurate error or warning
+messages when a stanza is used with a version of dune which does not support
+such a stanza.
+
+Create a config file to use in the following tests. We will initialize the 
+config with a version and a stanza incompatable with that version.
+
+  $ touch dune-config
+  $ cat >dune-config <<EOF
+  > (lang dune 1.0)
+  > (cache enabled)
+  > EOF
+
+Attempt to initialize a new project while an invaild stanza due to versioning
+exists in the config.
+
+  $ dune init proj test --config-file=dune-config
+  File "$TESTCASE_ROOT/dune-config", line 2, characters 0-15:
+  2 | (cache enabled)
+      ^^^^^^^^^^^^^^^
+  Error: 'cache' is only available since version 2.0 of the dune language.
+  Please update your dune config file to have (lang dune 2.0).
+  [1]
+
+Swap out the invaild stanza for another and try again.
+
+  $ sed -i -e '$s/.*/(cache-check-probability 0.5)/' dune-config
```

Comment:
This can be done with `echo "(cache-check-probability 0.5)" >> dune-config` which is a bit simpler to read and achieves a similar result.

Reply:
Here I am using sed to **swap** out the first invalid stanza *'(cache enabled)'*. `>>` has the effect of appending rather than swapping so the resulting error would be caused on the same *'(cache enabled)'* stanza from the first test instead of the new *cache-check-probability* stanza!

Reply:
Ahh sorry, I did not understand what you initially meant, thank you for clarifying! I've cleaned up the test cases let me know what you think.

Reply:
They have a good flow demonstrating the issue with the error and the fix, I like it.